test/cli/main.js

```diff
@@ -322,4 +444,55 @@ QUnit.module( "CLI Main", () => {
 		assert.equal( execution.stderr, "The `beforeEach` hook was called inside the wrong module. Instead, use hooks provided by the callback to the containing module. This will become an error in QUnit 3.0.", "The warning shows" );
 		assert.equal( execution.stdout, expectedOutput[ command ] );
 	} );
+
+	QUnit.module( "assert.expect failing conditions", () => {
+		QUnit.test( "mismatched expected assertions", async assert => {
+			const command = "qunit assert-expect/failing-expect.js";
+			try {
+				const result = await execute( command );
+				assert.pushResult( {
+					result: false,
+					actual: result.stdout
+				} );
+			} catch ( e ) {
+				assert.equal( e.code, 1 );
+				assert.equal( e.stderr, "" );
+				assert.notEqual( e.stdout.indexOf( "message: Expected 2 assertions, but 1 were run" ), -1, e.stdout );
+			}
+		} );
+
+		QUnit.test( "no assertions run - use expect(0)", async assert => {
+			const command = "qunit assert-expect/no-assertions.js";
+			try {
+				const result = await execute( command );
+				assert.pushResult( {
+					result: true,
+					actual: result.stdout
+				} );
+			} catch ( e ) {
+				assert.pushResult( {
+
+					// This isn't the *exact* the error message we want to see - it's been transformed
+					// by the execution to standardize on formatting. Should be improved.
```

Comment:
Is this still the case, or does your normalize fix make it match the real one? It does not look transformed to me.

Reply:
It was better, but still wrangling it. I'm left with a stack like this:
```
      stack: |
            at Object.<anonymous> (/qunit/test/cli/fixtures/assert-expect/no-assertions.js:1:7)
            at internal
            at run (/qunit/src/cli/run.js:79:5)
```
and that `run.js:79:5` is going to be fragile. That's not getting the "internal" escape from the regex since it starts with a forward-slash:

https://github.com/qunitjs/qunit/blob/main/test/cli/helpers/execute.js#L25
```js
.replace( / {2}at .+\([^/)][^)]*\)/g, "  at internal" )
```

I loosened that requirement, but that broke other verifications, so I left it alone here. I added some more straightforward comments; what I had was admittedly a bit cryptic, even to myself one week later :-/